src/cli/serve/serve.js

```diff
@@ -237,7 +247,7 @@ export default function (program) {
           silent: !!opts.silent,
           watch: !!opts.watch,
           repl: !!opts.repl,
-          basePath: !!opts.basePath,
+          basePath: opts.runExamples ? false : !!opts.basePath,
```

Comment:
The reason to use no base path here is that we can use links into our examples in our api documentation, much like we have the `View in Console` links, that links to `http://localhost:5601/kibana/demoSearch/...`.  We can link to these samples with a note that says, "be sure to running a local kibana with the `--run-examples` flag"